zilliqa/src/consensus.rs

```diff
@@ -1070,7 +1065,7 @@ impl Consensus {
 
     fn leader_for_view(&mut self, parent_hash: Hash, view: u64) -> Option<NodePublicKey> {
         if let Ok(Some(parent)) = self.get_block(&parent_hash) {
-            let leader = self.leader_at_block(&parent, view).unwrap();
+            let leader = self.leader_at_block(&parent).unwrap();
```

Comment:
Shouldn't we rather use the `leaderAtView()` function of the `Deposit` contract and pass in the `view` argument? The `view` parameter of the `leader_for_view()` function could refer to an arbitrary view number, not necessarily the current view.